File: backend/src/app/schemas/message.py
"""Schemas for messages"""
from typing import Optional, List

from fastapi_camelcase import CamelModel
from src.app.schemas.group import ReturnUserBasic, ReturnGroupBasic

class WriteMessage(CamelModel):
    """Schema to write a message"""
    message: str
    reply_id: Optional[int] = None


class ReturnMessageBasic(CamelModel):
    """Basic schema to return a message"""
    message_id: Optional[int]
    message: str
    sender_id: int
    sender: ReturnUserBasic


class ReturnMessage(ReturnMessageBasic):
    """Schema to return a message"""
    group_id: int
    group: ReturnGroupBasic
    reply_id: Optional[int] = None
    reply: Optional[ReturnMessageBasic] = None


class ReturnMessages(CamelModel):
    """Schema to return multiple messages"""
    messages: List[ReturnMessage]

File: frontend/frontend-chatapp/src/data/interfaces/message.ts
import { UserBasic } from "./user";
import { GroupBasic } from "./group";
export interface MessageBasic {
  message_id: number;
  message: string;
  sender_id: number;
  sender: UserBasic;
}

export interface Message {
  group_id: number;
  group: GroupBasic;
  reply_id: number | null;
  reply: MessageBasic | null;
}

export interface Messages {
  messages: Message[];
}
